src/Itk/ExchangeBundle/Services/ExchangeSoapClientService.php

```diff
@@ -94,36 +98,33 @@ public function __construct(
     }
 
     public function getAuthenticationToken() {
-        if ($this->token !== null && $this->tokenExpire !== null) {
-            $now = time();
+        $token = $this->cache->get(self::CACHE_KEY_TOKEN);
 
-            if ($now < $this->tokenExpire) {
-                return $this->token;
-            }
-        }
+        if ($token === null) {
+            $client = new Client();
 
-        $client = new Client();
+            $url = 'https://login.microsoftonline.com/'.$this->tenantId.'/oauth2/v2.0/token';
 
-        $url = 'https://login.microsoftonline.com/'.$this->tenantId.'/oauth2/v2.0/token';
+            $response = $client->post($url, [
+                    'form_params' => [
+                        'client_id' => $this->clientId,
+                        'client_secret' => $this->clientSecret,
+                        'scope' => 'https://outlook.office365.com/EWS.AccessAsUser.All',
+                        'username' => $this->username,
+                        'password' => $this->password,
+                        'grant_type' => 'password',
+                    ],
+                ]
+            );
 
-        $response = $client->post($url, [
-                'form_params' => [
-                    'client_id' => $this->clientId,
-                    'client_secret' => $this->clientSecret,
-                    'scope' => 'https://outlook.office365.com/EWS.AccessAsUser.All',
-                    'username' => $this->username,
-                    'password' => $this->password,
-                    'grant_type' => 'password',
-                ],
-            ]
-        );
+            $contents = json_decode($response->getBody()->getContents(), true);
```

Comment:
Do we have an try/catch when this i not json or when the `$client->post` do not return 200?